packages/mdc-icon-toggle/foundation.js

```diff
@@ -121,41 +149,85 @@ export default class MDCIconToggleFoundation extends MDCFoundation {
     }
   }
 
+  /**
+   * @param {string} dataAttr
+   * @return {!IconToggleState}
+   */
   parseJsonDataAttr_(dataAttr) {
     const val = this.adapter_.getAttr(dataAttr);
     if (!val) {
       return {};
     }
-    return JSON.parse(val);
+    return /** @type {!IconToggleState} */ (JSON.parse(val));
   }
 
+  /** @return {boolean} */
   isDisabled() {
     return this.disabled_;
   }
 
+  /** @param {boolean} isDisabled */
   setDisabled(isDisabled) {
     this.disabled_ = isDisabled;
 
     const {DISABLED} = MDCIconToggleFoundation.cssClasses;
     const {ARIA_DISABLED} = MDCIconToggleFoundation.strings;
 
     if (this.disabled_) {
-      this.savedTabIndex = this.adapter_.getTabIndex();
+      this.savedTabIndex_ = this.adapter_.getTabIndex();
       this.adapter_.setTabIndex(-1);
       this.adapter_.setAttr(ARIA_DISABLED, 'true');
       this.adapter_.addClass(DISABLED);
     } else {
-      this.adapter_.setTabIndex(this.savedTabIndex);
+      this.adapter_.setTabIndex(this.savedTabIndex_);
       this.adapter_.rmAttr(ARIA_DISABLED);
       this.adapter_.removeClass(DISABLED);
     }
   }
 
+  /** @return {boolean} */
   isKeyboardActivated() {
     return this.isHandlingKeydown_;
   }
 }
 
-function isSpace({key, keyCode}) {
-  return key && key === 'Space' || keyCode === 32;
+/**
+ * @typedef {!{
+ *   key: string,
+ *   keyCode: number
+ * }}
+ */
+export let KeyboardKey;
+
+/**
+ * @param {KeyboardKey} keyboardKey
+ * @return {boolean}
+ */
+function isSpace(keyboardKey) {
+  return !!keyboardKey.key && keyboardKey.key === 'Space' || keyboardKey.keyCode === 32;
 }
+
+
+/** @record */
+class IconToggleState {}
+
+/**
+ * The aria-label value of the icon toggle, or undefined if there is no aria-label.
+ * @type {string|undefined}
```

Comment:
These two lines can be collapsed into one: `@export {string|undefined}`. Ditto below